src/dist/signatures.rs

```diff
@@ -56,20 +56,12 @@ impl VerificationHelper for Helper<'_> {
         for layer in structure.into_iter() {
             match layer {
                 MessageLayer::SignatureGroup { results } => {
-                    for result in results {
-                        match result {
-                            Ok(GoodChecksum { ka, .. }) => {
-                                // A good signature!  Find the index
-                                // of the singer key and return
-                                // success.
-                                self.index = self.certs.iter().position(|c| c.cert() == ka.cert());
-                                assert!(self.index.is_some());
-                                return Ok(());
-                            }
-                            _ => {
-                                // We ignore any errors.
-                            }
-                        }
+                    for GoodChecksum { ka, .. } in results.into_iter().flatten() {
+                        // A good signature!  Find the index
+                        // of the singer key and return
+                        // success.
+                        self.index = self.certs.iter().position(|c| c.cert() == ka.cert());
+                        assert!(self.index.is_some());
```

Comment:
This loses the nice explicit indicator that we are ignoring errors during signature verification, which is an open question for us.  We shouldn't hide that.  If you're going to do this refactor then please ensure there's a comment to the effect that the `.flatten()` causes us to ignore errors for now.  Also fix the comment `singer` -> `signer` please :D